Sources/LiveKit/Auth/ConnectionCredentials.swift

```diff
@@ -0,0 +1,269 @@
+        /// - SeeAlso: [Room Configuration Documentation](https://docs.livekit.io/home/get-started/authentication/#room-configuration) for more info.
+        let roomConfiguration: RoomConfiguration?
+
+        public init(
+            roomName: String? = nil,
+            participantName: String? = nil,
+            participantIdentity: String? = nil,
+            participantMetadata: String? = nil,
+            participantAttributes: [String: String]? = nil,
+            roomConfiguration: RoomConfiguration? = nil
+        ) {
+            self.roomName = roomName
+            self.participantName = participantName
+            self.participantIdentity = participantIdentity
+            self.participantMetadata = participantMetadata
+            self.participantAttributes = participantAttributes
+            self.roomConfiguration = roomConfiguration
+        }
+    }
+
+    /// Response containing the credentials needed to connect to a room.
+    public struct Response: Decodable, Sendable {
+        /// The WebSocket URL for the LiveKit server.
+        let serverUrl: URL
+        /// The JWT token containing participant permissions and metadata.
+        let participantToken: String
+
+        public init(serverUrl: URL, participantToken: String) {
+            self.serverUrl = serverUrl
+            self.participantToken = participantToken
+        }
+    }
+
+    public typealias Options = Request
+    public typealias Literal = Response
+}
+
+// MARK: - Provider
+
+/// Protocol for types that can provide connection credentials.
+/// Implement this protocol to create custom credential providers (e.g., fetching from your backend API).
+public protocol CredentialsProvider: Sendable {
+    func fetch(_ request: ConnectionCredentials.Request) async throws -> ConnectionCredentials.Response
+}
+
+/// `ConnectionCredentials.Literal` contains a single set of credentials, hard-coded or acquired from a static source.
+/// - Note: It does not support refresing credentials.
+extension ConnectionCredentials.Literal: CredentialsProvider {
+    public func fetch(_: ConnectionCredentials.Request) async throws -> ConnectionCredentials.Response {
+        self
+    }
+}
+
+// MARK: - Token Server
+
+/// Protocol for token servers that fetch credentials via HTTP requests.
+/// Provides a default implementation of `fetch` that can be used to integrate with custom backend token generation endpoints.
+/// - Note: The response is expected to be a `ConnectionCredentials.Response` object.
+public protocol TokenServer: CredentialsProvider {
+    /// The URL endpoint for token generation.
+    var url: URL { get }
+    /// The HTTP method to use (defaults to "POST").
+    var method: String { get }
+    /// Additional HTTP headers to include with the request.
+    var headers: [String: String] { get }
+}
+
+public extension TokenServer {
+    var method: String { "POST" }
```

Comment:
Something like this might be worth doing on the wek too - Lukas had been referring to this as `ConnectionCredentials.Standard`. I realize it's a protocol and not a real object definition, but I guess what I'm wondering is if maybe it's worth upgrading it given ben's vision of "one request / response standard that can be used everywhere".

Reply:
Ah, I see. Yea that seems decent then, good point.